core/ir/aarch64/encode.c

```diff
@@ -219,7 +219,8 @@ encoding_possible(decode_info_t *di, instr_t *in, const instr_info_t *ii)
 void
 decode_info_init_for_instr(decode_info_t *di, instr_t *instr)
```

Comment:
The current logic doesn't need to use the passed-in instr at all. Can't we do a simple brace init instead of this memset? Or maybe do `di.check_reachable = false` also on RISC-V (like we do for AArch64), which would add less overhead?

Reply:
My concern is that initializing using memset is slower than initializing just the one field that's required, which may become apparent on frequent calls.

Reply:
For decoder related implementations (where code can potentially be invoked for each instr), I feel it may be better to just set check_reachable instead of memset.

Reply:
At least for now, `check_reachable` is the only field in `decode_info_t` for both Aarch64/RISC-V. How about just delete the `memset(di, 0, sizeof(*di));` line?

Reply:
Deleted the `memset`s in both Aarch64 and RISC-V.

Reply:
Thanks.